### ISSUE TITLE
Native TLS/SSL bindings for Jai

### ISSUE THREAD

**Issue by frikky** [smari/jai-crypto#1]
Hey!

My first day playing with the Jai compiler today, and got a HTTP socket client application working with the inputs and such I wanted. I started extending it to support TLS and was wondering if you had worked on anything similar in the last years since making this package?

Googling doesn't help much so I'm asking around :)

**Comment by smari** [comment]
Hi! Nope, haven't done any SSL/TLS or HTTPS stuff yet. Would like to get there at some point, currently I've got my HTTP server hiding behind an NGINX server that provides that (and some additional security).

This package is kind of sad at the moment. It only really supports the SHA2 family, and isn't even doing that in the most sensible possible way. I've been thinking of unlisting it, but the better thing to do would be to give it the love it deserves - a few block ciphers and block cipher modes, some more hash functions, etc. 

That said, this is not going to be the right place for TLS/SSL -- at least, I don't currently think so.

**Comment by frikky** [comment]
> Hi! Nope, haven't done any SSL/TLS or HTTPS stuff yet. Would like to get there at some point, currently I've got my HTTP server hiding behind an NGINX server that provides that (and some additional security).
> 
> This package is kind of sad at the moment. It only really supports the SHA2 family, and isn't even doing that in the most sensible possible way. I've been thinking of unlisting it, but the better thing to do would be to give it the love it deserves - a few block ciphers and block cipher modes, some more hash functions, etc.
> 
> That said, this is not going to be the right place for TLS/SSL -- at least, I don't currently think so.

Gotcha, seems like the kinds of workarounds I'm used to as well. It doesn't really help when building a client tho.

Anyway, I think building it here or in another repo wouldn't really matter, as the goal would probably be to have a native Crypto package (I would assume). Just depends on who is good enough at crypto programming to actually implement it well 🔥

**Comment by smari** [comment]
Agreed. I think making bindings for TLS/SSL (OpenSSL probably) would be the sensible thing to do for now. Ping me on the Discord if you want help getting started with that, if that's what you're after. :-)
